x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/application/EnterpriseSearchFeatureSetUsage.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.application;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.core.XPackFeatureSet;
+import org.elasticsearch.xpack.core.XPackField;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+public class EnterpriseSearchFeatureSetUsage extends XPackFeatureSet.Usage {
+
+    public static final String SEARCH_APPLICATIONS = "search_applications";
+    private final Map<String, Object> searchApplicationsStats;
```

Comment:
Shouldn't we use a specific class for the `SearchApplicationStats`? That would allow us to understand what stats are gathered, and that's what other examples like [RemoteClusterFeatureSetUsage](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/RemoteClusterFeatureSetUsage.java) do...

I know for now it's just a little information, but it would be easier to extend that way...

Reply:
this should be fine for what we are doing right now, we can always add this later, it's not required for this PR IMO.
also we can find other examples that just use a `Map<String, Object>` and they seem to work fine:

https://github.com/elastic/elasticsearch/blob/ba69ff661078ab0a65240f08199f28c95a22c2b6/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/MachineLearningFeatureSetUsage.java#L34-L37

Reply:
Fine with me then - I'm such a fan of strong typing! If you're all ok I'm as well 🙂